ddl/partition.go

```diff
@@ -305,72 +305,62 @@ func checkRangePartitioningKeysConstraints(ctx sessionctx.Context, expr ast.Expr
 		partkeys = append(partkeys, col.ColName.L)
 	}
 
-	switch expr.(type) {
-	case *ast.ColumnNameExpr:
-		// Only one column name expression.
-		if checkConstraintIncludePartKey(partkeys, priKeys, false) || checkConstraintIncludePartKey(partkeys, multipleKeys, false) {
-			return ErrUniqueKeyNeedAllFieldsInPf.GenByArgs("PRIMARY KEY")
-		}
-	default:
-		// Other expression, with multiple column names.
-		if checkConstraintIncludePartKey(partkeys, priKeys, true) || checkConstraintIncludePartKey(partkeys, multipleKeys, true) {
-			return ErrUniqueKeyNeedAllFieldsInPf.GenByArgs("PRIMARY KEY")
-		}
-
+	// Checks that the partitioning key is included in the constraint.
```

Comment:
That's enough for the rest of the function:
```
if !checkConstraintIncludePartKey(partKeys, consColNames) {
    return ErrUniqueKeyNeedAllFieldsInPf
}
return nil
```
We don't care for the error message, as long as the error code is right.